sonarts-core/src/rules/noIgnoredInitialValueRule.ts

```diff
@@ -0,0 +1,124 @@
+import { SonarRuleMetaData } from "../sonarRule";
+import { is, FUNCTION_LIKE, descendants, ancestorsChain } from "../utils/navigation";
+import { LiveVariableAnalyzer, LVAReturn } from "../symbols/lva";
+
+export class Rule extends tslint.Rules.TypedRule {
+  public static metadata: SonarRuleMetaData = {
+    ruleName: "no-ignored-initial-value",
+    description: "Function parameters, caught exceptions and foreach variables' initial values should not be ignored",
+    optionsDescription: "",
+    options: null,
+    rspecKey: "RSPEC-1226",
+    type: "functionality",
+    typescriptOnly: false,
+  };
+
+  public static formatMessage(deadIdentifier: ts.Identifier) {
+    return `Introduce a new variable or use its initial value before reassigning "${deadIdentifier.getText()}".`;
+  }
+
+  public applyWithProgram(sourceFile: ts.SourceFile, program: ts.Program): tslint.RuleFailure[] {
+    const symbols = SymbolTableBuilder.build(sourceFile, program);
+    return this.applyWithWalker(new Walker(sourceFile, this.getOptions(), program, symbols));
+  }
+}
+
+class Walker extends tslint.ProgramAwareRuleWalker {
+  private lva: LiveVariableAnalyzer;
+
+  constructor(
+    sourceFile: ts.SourceFile,
+    options: tslint.IOptions,
+    program: ts.Program,
+    private readonly symbols: SymbolTable,
+  ) {
+    super(sourceFile, options, program);
+    this.lva = new LiveVariableAnalyzer(this.symbols);
+  }
+
+  protected visitNode(node: ts.Node): void {
+    if (is(node, ...FUNCTION_LIKE)) {
+      const functionLike = node as ts.FunctionLikeDeclaration;
+      const lvaReturn = this.lva.analyzeFunction(functionLike);
+      this.check(functionLike.body!, lvaReturn, ...functionLike.parameters);
+    } else if (is(node, ts.SyntaxKind.CatchClause)) {
+      const catchClause = node as ts.CatchClause;
+      const lvaReturn = this.lva.analyze(catchClause.block, catchClause.block.statements);
+      this.check(catchClause.block, lvaReturn, catchClause.variableDeclaration);
+    } else if (is(node, ts.SyntaxKind.ForInStatement, ts.SyntaxKind.ForOfStatement)) {
+      const iterationStatement = node as ts.IterationStatement;
+      const lvaReturn = this.lva.analyze(iterationStatement, [iterationStatement.statement]);
+      this.check(iterationStatement.statement, lvaReturn, (iterationStatement as ts.ForInStatement).initializer);
+    }
+
+    super.visitNode(node);
+  }
+
+  private check(root: ts.Node, lvaReturn?: LVAReturn,  ...nodesToCheck: ts.Node[]) {
+    if (!lvaReturn) {
+      return;
+    }
+    const { cfg, blockAvailableReads } = lvaReturn;
+    const symbolsLiveAtStart = blockAvailableReads.get(cfg.start)!;
+
+    nodesToCheck.forEach(parameter => {
+      descendants(parameter)
+        .filter(descendant => is(descendant, ts.SyntaxKind.Identifier))
+        .forEach(descendant => {
+          const identifier = descendant as ts.Identifier;
+          const symbol = this.getTypeChecker().getSymbolAtLocation(identifier);
```

Comment:
@vilchik-elena you could use `this.symbols.getUsage(identifier)`, just to avoid using both the symbol table and the type checker